src/info/git/mod.rs

```diff
@@ -172,20 +174,29 @@ type ChurnPair = (NumberOfCommitsByFilepath, usize);
 
 fn get_churn_channel(
     repo: &gix::Repository,
+    mailmap: &gix::mailmap::Snapshot,
+    bot_regex_pattern: Option<MyRegex>,
     has_commit_graph_traversal_ended: &Arc<AtomicBool>,
     total_number_of_commits: &Arc<AtomicUsize>,
     max_churn_pool_size: Option<usize>,
 ) -> Result<(JoinHandle<Result<ChurnPair>>, Sender<ObjectId>)> {
     let (tx, rx) = channel::<gix::hash::ObjectId>();
     let thread = std::thread::spawn({
         let repo = repo.clone();
+        let mailmap = mailmap.clone();
+        let bot_regex_pattern = bot_regex_pattern.clone();
         let has_commit_graph_traversal_ended = has_commit_graph_traversal_ended.clone();
         let total_number_of_commits = total_number_of_commits.clone();
         move || -> Result<_> {
             let mut number_of_commits_by_file_path = NumberOfCommitsByFilepath::new();
             let mut number_of_diffs_computed = 0;
             while let Ok(commit_id) = rx.recv() {
                 let commit = repo.find_object(commit_id)?.into_commit();
+                if bot_regex_pattern.is_some()
```

Comment:
This extra check shouldn't be needed if `is_bot_commit()` would exit early. I agree that all the hassle might go away if `is_bot_commit()` would take the regex without an `Option` right away.

Reply:
Done, moved the check inside `is_bot_commit`